src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java

```diff
@@ -52,6 +51,35 @@ public void processDelete(Message<ResourceChangedData> resourceChangedMessage) {
         logger.trace(String.format("Resource changed message for deleted event of kind %s "
                 + "for company number %s retrieved", payload.getResourceKind(), companyNumber));
 
+        final ApiResponse<CompanyProfile> response =
+                companyProfileService.getCompanyProfile(logContext, companyNumber);
+        logger.trace(String.format("Retrieved company profile for company number %s: %s",
+                companyNumber, response.getData()));
+        handleResponse(HttpStatus.valueOf(response.getStatusCode()), logContext,
+                "Response from GET call to company profile api", logMap, logger);
+        var data = response.getData().getData();
+        var links = data.getLinks();
+
+        if (links == null) {
+            logger.trace(String.format("Company profile with company number %s,"
+                            + " does not contain insolvency links, will not perform patch",
+                    companyNumber));
+            return;
+        }
+
+        data.setLinks(null);
```

Comment:
We need only to remove the insolvency link and not all the links. Please address this.

Reply:
changes done